### PR TITLE
Add it for sharding mix encrypt

### PR DESCRIPTION
**Pull request by tuichenchuxin**
For #19691

Changes proposed in this pull request:
- Add it for sharding mix encrypt

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/data/actual/databases.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<databases>
+    <database>encrypt_write_ds_0</database>
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/data/actual/databases.xml:20]: Why exist read and write datasources in the sharding and encrypt scenarios?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/proxy/conf/mysql/config-sharding-and-encrypt.yaml [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+databaseName: sharding_and_encrypt
+
+dataSources:
+  encrypt_write_ds_0:
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/proxy/conf/mysql/config-sharding-and-encrypt.yaml:21]: Maybe we should consider to rename these datasource name.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/data/actual/databases.xml (hunk 1 above)
Comment (tuichenchuxin) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/data/actual/databases.xml]: Fixed. Thank you.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/proxy/conf/mysql/config-sharding-and-encrypt.yaml (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-suite/src/test/resources/env/scenario/sharding_and_encrypt/proxy/conf/mysql/config-sharding-and-encrypt.yaml]: Done. Thank you

**Review by strongduanmu — APPROVED** [review]
Looks perfect, merged.

**Pull request merged**